### PR TITLE
Add filtering for events reported to Sentry

### PR DESCRIPTION
**Pull request by raccube**

### CODE REVIEW

**Review by pixeldesu — APPROVED** [review]

File: config/initializers/sentry.rb [hunk 1]
```diff
@@ -8,4 +8,22 @@
   # of transactions for performance monitoring.
   # We recommend adjusting this value in production
   config.traces_sample_rate = 0.25
+
+  exception_fingerprints = {
+    Excon::Error::ServiceUnavailable => 'external-service',
+    Twitter::Error::InternalServerError => 'external-service',
+  }
+  config.before_send do |event, hint|
```
Comment (nilsding) [inline config/initializers/sentry.rb]: isn't it

```suggestion
  config.before_send = lambda do |event, hint|
```

File: config/initializers/sentry.rb [hunk 2]
```diff
@@ -8,4 +8,22 @@
...
+    if hint[:exception].is_a?(Errors::Base)
+      # These are used for user-facing errors, not when something goes wrong
+      nil
```
Comment (nilsding) [inline config/initializers/sentry.rb]: use `next` to return from a block early
```suggestion
      next
```

File: config/initializers/sentry.rb [hunk 3]
```diff
@@ -8,4 +8,22 @@
...
+    Twitter::Error::InternalServerError => 'external-service',
+  }
+  config.before_send = lambda do |event, hint|
+    if hint[:exception].is_a?(Errors::Base)
+      # These are used for user-facing errors, not when something goes wrong
+      nil
+    end
+
+    exception_class = hint[:exception].class.name
+    if exception_fingerprints.key?(exception_class)
```
Comment (nilsding) [inline config/initializers/sentry.rb]: `class.name` returns a String, so `#key?` will never match anything

**Review by nilsding — CHANGES_REQUESTED** [review]
beep boop github needs a comment to request changes

**Review by nilsding — APPROVED** [review]

**Pull request merged**